core/modules/image/image.tokens.inc

```diff
@@ -0,0 +1,327 @@
+        }
+      }
+    }
+
+    $images_style_data = &backdrop_static(__FUNCTION__);
+
+    if ($type == 'image-field' && !empty($data['image-field'])) {
+      foreach ($tokens as $token => $original) {
+        $output = array();
+
+        // The token will be in the format "[style]:[attribute]" so split into
+        // the individual pieces.
+        $explode = explode(':', $token);
+        $type = isset($explode[0]) ? $explode[0] : FALSE;
+        $attribute = isset($explode[1]) ? $explode[1] : FALSE;
+
+        // Special handling for the 'first', 'second', 'third', and 'last' tokens.
+        // Support for index of multi-value fields.
+        if (in_array($attribute, array('first', 'second', 'third', 'last')) || is_numeric($attribute)) {
+          switch ($attribute) {
+            case 'first':
+              $index = 0;
+              break;
+
+            case 'second':
+              $index = 1;
+              break;
+
+            case 'third':
+              $index = 2;
+              break;
+
+            case 'last':
+              $index = sizeof($data['image-field']) - 1;
+              break;
+
+            default:
+              $index = $attribute;
+          }
+
+          if (isset($data['image-field'][$index]) && !empty($data['image-field'][$index]['uri'])) {
+            if (isset($explode[3]) && $explode[3] === 'relative') {
+              $url  = parse_url(image_style_url($type, $data['image-field'][$index]['uri']));
+              $output[] = $url['path'];
+            }
+            else {
+              $output[] = image_style_url($type, $data['image-field'][$index]['uri']);
+            }
+
+            // Generate image if does not exist.
+            $styled_uri = image_style_path($type, $data['image-field'][$index]['uri']);
+            _image_generate_image($styled_uri, $type, $data['image-field'][$index]['uri']);
+          }
+
+          $replacements[$original] = implode('', $output);
+        }
+
+        // Other tokens.
+        else {
+          foreach ($data['image-field'] as $field) {
+            if (isset($attribute) && !empty($attribute)) {
+              // Generate image if does not exist.
+              $styled_uri = image_style_path($type, $field['uri']);
+              _image_generate_image($styled_uri, $type, $field['uri']);
+
+              switch ($attribute) {
+                case 'render':
+                  $output[] = theme('image_formatter', array('item' => $field, 'image_style' => $type));
+                  break;
+
+                case 'mimetype':
+                case 'width':
+                case 'height':
+                case 'extension':
+                case 'filesize':
+                  // If the styled image's information wasn't loaded before then
+                  // load it now; as this process is a little heavy, the
+                  // information will be statically cached to avoid needing to
+                  // reload the details for another token.
```

Comment:
```suggestion
                  // If the styled image's information wasn't loaded before, then
                  // load it now; as this process is a little heavy, the
                  // information will be statically cached, to avoid needing to
                  // reload the details for another token.
```

Reply:
the last added comma isn't grammatically correct, but I like the first one. I'll merge this PR then remove the extra comma :)